app/src/main/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestFragment.java

```diff
@@ -23,24 +21,32 @@
     HintLabelEditText contestNameField;
     @BindView(R.id.trophy_icon)
     ImageView trophyIcon;
+    EditContestContract contestEditorActivity;
 
     @Override
-    public void onCreate(@Nullable Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    protected int getLayoutResourceId() {
+        return R.layout.fragment_edit_contest_name;
     }
 
     @Override
-    protected int getLayoutResourceId() {
-        return R.layout.fragment_edit_contest_name;
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        contestEditorActivity = (EditContestContract) getActivity();
     }
 
     @OnTextChanged(R.id.hint_label_edit_text)
     public final void onTextChanged(CharSequence newName) {
-        if(TextUtils.isEmpty(newName)){
-            trophyIcon.setVisibility(VISIBLE);
-        }else{
+        presenter.onTextChanged(newName);
+    }
+
+    @Override
+    public void setNextEnabled(boolean enabled) {
+        if (enabled) {
```

Comment:
one neat way to wrap these that Glen likes to use a lot is:

int visibility = (enabled) ? VISIBLE : GONE
trophyIcon.setVisiblity(visibility)

it makes things a bit "DRYer" if you need to change stuff later.